e2e/src/componentsController.spec.ts

```diff
@@ -636,4 +646,462 @@ describe('Components Controller', () => {
+          description: '',
+          requireCredit_card: true,
+          priceInCents: BigInt(1000),
+          interval: 1,
+          intervalUnit: IntervalUnit.Month,
+        },
+      };
+
+      const price: Price = {
+        starting_quantity: '1',
+        ending_quantity: '',
+        unit_price: '1',
+        startingQuantity: '',
+        unitPrice: '',
+      };
+
+      const productFamilyId = productFamily?.id || 0;
+
+      const createComponentPricePointRequestPricePoint: CreateComponentPricePointRequestPricePoint =
+        {
+          name: `price-point-name-${uid()}`,
+          handle: `price-point-handle-${uid()}`,
+          prices: [price],
+          pricing_scheme: 'per_unit',
+          tax_included: false,
+          pricingScheme: PricingScheme.PerUnit,
+          overagePricing: [],
+        };
+
+      const pricePoint: CreateComponentPricePointRequest = {
+        pricePoint: createComponentPricePointRequestPricePoint,
+      };
+
+      const meteredComponent: CreateMeteredComponent = {
+        meteredComponent: {
+          productFamilyId,
+          name: `promote-component-price-to-default-component-name${uid()}`,
+          description: 'test',
+          quantity: 1,
+          unitName: 'test',
+          price: 1,
+          handle: uid(),
+          pricingScheme: PricingScheme.PerUnit,
+          unitPrice: 1,
+        },
+      };
+
+      const { component } = await prepareDataForComponentResponse(
+        newProductPayload,
+        productFamilyId.toString(),
+        meteredComponent
+      );
+
+      await componentPricePointsController.createComponentPricePoint(
+        component.id || 0,
+        pricePoint
+      );
+
+      const invalidPricePointId = 0;
+      const invalidComponentId = 0;
+
+      const promoteComponentPricePointToDefaultResponse =
+        componentPricePointsController.promoteComponentPricePointToDefault(
+          invalidComponentId,
+          invalidPricePointId
+        );
+
+      expect(promoteComponentPricePointToDefaultResponse).rejects.toThrow();
+
+      await promoteComponentPricePointToDefaultResponse.catch((reason) => {
+        expect(reason.statusCode).toBe(404);
+      });
+    });
+    it('should not promote component price to default with invalid credentials', async () => {
+      const newProductPayload: CreateOrUpdateProductRequest = {
+        product: {
+          name: `promote-component-price-to-default-name-${uid()}`,
+          handle: `promote-component-price-to-default-handle-${uid()}`,
+          description: '',
+          requireCredit_card: true,
+          priceInCents: BigInt(1000),
+          interval: 1,
+          intervalUnit: IntervalUnit.Month,
+        },
+      };
+
+      const price: Price = {
+        starting_quantity: '1',
+        ending_quantity: '',
+        unit_price: '1',
+        startingQuantity: '',
+        unitPrice: '',
+      };
+
+      const productFamilyId = productFamily?.id || 0;
```

Comment:
you just described it above as a string, so what's the real type?

Reply:
productFamily.id is a number, but I need to proivde it as a string to requests in my prepareDataForComponentResponse method

Reply:
you are casting it via toString in there anyways

Reply:
ok make sense! So I will revert changes, and make it as a number, but cast it to string only in those requests

Reply:
done